### PR TITLE
#735 Fix and re-enable Stackoverflow ZIMs

### LINKED ISSUE #735: Revert #734 to show stack exchange zims
#### Feature Description

Re-enable stack exchange zim files once they work again.

### PR DESCRIPTION
**Pull request by mhutti1**
Fixes #735 

Adds code to handle relative urls in the case of redirection correctly, so that they are parsed relative to the redirected URL. This is needed to re add stackoverflow ZIM files which have been hidden as they don't display correctly.

### CODE REVIEW

**Comment by kelson42** [comment]
@mhutti1 @macgills  New libzim has been released. What is next for this PR?

**Comment by macgills** [comment]
I can take it over, merge develop in, test. Or Isaac can do it. But it is now unblocked.
The stackoverflow zims must also have their filter removed

**Comment by mhutti1** [comment]
@macgills feel free to take this over and merge if we have kiwix-lib at the right version now.

File: app/src/test/java/org/kiwix/kiwixmobile/zim_manager/ZimManageViewModelTest.kt [hunk 1]
```diff
@@ -337,9 +337,9 @@ class ZimManageViewModelTest {
       )
       expectNetworkDbAndDefault(
         listOf(
-          Book().apply { language = "eng" },
-          Book().apply { language = "eng" },
-          Book().apply { language = "fra" }
+          book(language = "eng"),
+          book(language = "eng"),
```
Comment (kelson42) [inline app/src/test/java/org/kiwix/kiwixmobile/zim_manager/ZimManageViewModelTest.kt]: @macgills I don't understand much what is going on, but this duplicate line seems really weird too me.

Comment (macgills) [reply app/src/test/java/org/kiwix/kiwixmobile/zim_manager/ZimManageViewModelTest.kt]: this shows that when there a multiple of the same language in the network result it increments the occurrence of the language. if you look at the full test, specifically the verify
```
 verify {
        newLanguagesDao.insert(
          listOf(
            dbLanguage.copy(occurencesOfLanguage = 2),
            Language(
              active = false,
              occurencesOfLanguage = 1,
              language = "fra",
              languageLocalized = "",
              languageCode = "",
              languageCodeISO2 = ""
            )
          )
        )
      }
```

Comment (macgills) [reply app/src/test/java/org/kiwix/kiwixmobile/zim_manager/ZimManageViewModelTest.kt]: The `.copy` method is a great little addition to a `data class` in kotlin. it allows you to create a new copy of an object but the method takes named parameters for any field of the object so here I only wanted to update `occurencesOfLanguage ` in my new object

**Pull request merged**